spec/models/tag_prompt_deployment_spec.rb

```diff
@@ -1,10 +1,69 @@
 describe TagPromptDeployment do
 	let(:tag_dep) { TagPromptDeployment.new id: 1, tag_prompt: tp, tag_prompt_id: 1, question_type: "Criterion", answer_length_threshold: 5 }
+	let(:tag_dep1) { TagPromptDeployment.new id: 1, tag_prompt: tp, tag_prompt_id: 1, question_type: "Criterion", answer_length_threshold: nil, assignment_id: 1, assignment: assignment , questionnaire: questionaire}
 	let(:tp) { TagPrompt.new(prompt: "test prompt", desc: "test desc", control_type: "Checkbox") }
+	let(:team) {Team.new(id: 1)}
+	let(:assignment) {Assignment.new(id: 1)}
+	let(:questionaire) {Questionnaire.new(id: 1,name: 'question1') }
+	let(:rp) {Response.new(map_id: 1, round: 1, additional_comment: "improvement scope") }
+	let(:responses) {Response.new(map_id:[1,2], round: [1, 1], additional_comment: ["improvement scope",'through comments']) }
+	let(:question) {Question.new(questionnaire_id: 1, type: 'tagging')}
+	let(:answers) {Answer.new(id:[1,2,3], question_id:[1,1,1], answer: [3,3,3], comments: ['comment', 'comment is lengthy', 'comment is too lengthy'], response_id: [241, 241, 241])}
+	let(:answers_one) {Answer.new(id:[1], question_id:[1], answer: [3], comments: ['comment'], response_id: [241])}
+
 	describe '#tag_prompt' do
 		it 'returns the associated tag prompt with the deployment' do
 			allow(TagPrompt).to receive(:find).with(1).and_return(tp)
 			expect(tag_dep.tag_prompt).to be(tp)
 		end
 	end
-end
\ No newline at end of file
+#get_number_of_taggable_answers calculates total taggable answers assigned to an user who participated in "tag review assignment".
+	describe '#get_number_of_taggable_answers' do
+		before(:each) do
+			questions_ids = double(1)
+			responses_ids = double(241)
```

Comment:
Line 23 and 24 are repeated at two places lines (38, 39) and lines (47, 48). Are all 3 needed?

Reply:
Yes, they have required else it gives error.